### PR TITLE
[#2896] improve(IT): Add e2e test case for verifying fileset schema level and kafka topic level

### PR DESCRIPTION
**Pull request by LauraXia123**
…a topic level

### What changes were proposed in this pull request?
Improvement of fileset and kafka test case

### Why are the changes needed?

Fix: #2896

### Does this PR introduce _any_ user-facing change?
N/A

### How was this patch tested?
image

### CODE REVIEW

File: integration-test/src/test/java/com/datastrato/gravitino/integration/test/web/ui/CatalogsPageTest.java [hunk 1]
```diff
@@ -92,8 +106,38 @@ public static void before() throws Exception {
     hdfsUri = trinoITContainers.getHdfsUri();
     mysqlUri = trinoITContainers.getMysqlUri();
     postgresqlUri = trinoITContainers.getPostgresqlUri();
+
+    // containerSuite.startKafkaContainer();
```
Comment (yuqi1129) [inline integration-test/src/test/java/com/datastrato/gravitino/integration/test/web/ui/CatalogsPageTest.java:110]: Why do you comment out the Kafka container? I notice that CI fails as the Kafka server was not found.

Comment (LauraXia123) [reply integration-test/src/test/java/com/datastrato/gravitino/integration/test/web/ui/CatalogsPageTest.java:110]: It's a draft commit of startKafkaContainer, cannot found the kafka image when call the startKafkaContainer.

**Comment by jerryshao** [comment]
@xunliu @mchades can you please help to review when you have time.

File: integration-test/src/test/resources/run
Comment (mchades) [inline integration-test/src/test/resources/run:1]: This file is copied from `catalogs/catalog-kafka/src/test/resources/run`, I‘m not sure can we reuse it or the current approach is also acceptable?

File: integration-test/src/test/java/com/datastrato/gravitino/integration/test/web/ui/CatalogsPageTest.java [hunk 2]
```diff
@@ -111,6 +155,92 @@ void createTableAndColumn(
             properties);
   }
 
+  /**
+   * Retrieves the default base location for the given schema name.
+   *
+   * @param schemaName The name of the schema.
+   * @return The default HDFS storage location for the schema.
+   */
+  private static String defaultBaseLocation(String schemaName) {
```
Comment (ch3yne) [inline integration-test/src/test/java/com/datastrato/gravitino/integration/test/web/ui/CatalogsPageTest.java:164]: Is it more reasonable to use `setDefaultBaseLocation` for this variable name?

File: integration-test/src/test/java/com/datastrato/gravitino/integration/test/web/ui/CatalogsPageTest.java [hunk 3]
```diff
@@ -464,6 +621,143 @@ public void testTreeNodeRefresh() throws InterruptedException {
 
   @Test
   @Order(20)
+  public void testViewFilesetCatalog() throws InterruptedException {
```
Comment (ch3yne) [inline integration-test/src/test/java/com/datastrato/gravitino/integration/test/web/ui/CatalogsPageTest.java:624]: `testViewFilesetCatalog` and `testViewKafkaCatalog` are best placed in the test process above.
I think we should avoid adding other operations in a single function test to make the test process more integrated.

File: integration-test/src/test/java/com/datastrato/gravitino/integration/test/web/ui/CatalogsPageTest.java [hunk 4]
```diff
@@ -464,6 +621,143 @@ public void testTreeNodeRefresh() throws InterruptedException {
...
+            ICEBERG_CATALOG_NAME,
+            MYSQL_CATALOG_NAME,
+            PG_CATALOG_NAME,
+            FILESET_CATALOG_NAME,
+            KAFKA_CATALOG_NAME,
+            SCHEMA_NAME,
+            TOPIC_NAME);
+    Assertions.assertTrue(catalogsPage.verifyTreeNodes(treeNodes));
+    // 7. click fileset tree node
+    String filesetNode =
```
Comment (ch3yne) [inline integration-test/src/test/java/com/datastrato/gravitino/integration/test/web/ui/CatalogsPageTest.java:737]: The variable name should be `kafkaNode`?

File: integration-test/src/test/java/com/datastrato/gravitino/integration/test/web/ui/CatalogsPageTest.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (LauraXia123) [reply integration-test/src/test/java/com/datastrato/gravitino/integration/test/web/ui/CatalogsPageTest.java:164]: It's up to the function from HadoopCatalogIT.java

File: integration-test/src/test/java/com/datastrato/gravitino/integration/test/web/ui/CatalogsPageTest.java (hunk 4 above)
Comment (LauraXia123) [reply integration-test/src/test/java/com/datastrato/gravitino/integration/test/web/ui/CatalogsPageTest.java:737]: It should be topicNode, I have changed it correct.

File: integration-test/src/test/java/com/datastrato/gravitino/integration/test/web/ui/CatalogsPageTest.java (hunk 3 above)
Comment (LauraXia123) [reply integration-test/src/test/java/com/datastrato/gravitino/integration/test/web/ui/CatalogsPageTest.java]: done

File: integration-test/src/test/resources/run
Comment (xunliu) [inline integration-test/src/test/resources/run:1]: I didn't find a space to use this `run` file.

File: integration-test/src/test/java/com/datastrato/gravitino/integration/test/web/ui/CatalogsPageTest.java
Comment (xunliu) [inline integration-test/src/test/java/com/datastrato/gravitino/integration/test/web/ui/CatalogsPageTest.java:1]: I think we need split a new `xxxCatalogPageTest.java` to test kafka or fileset catalogs.

File: integration-test/src/test/resources/run
Comment (LauraXia123) [reply integration-test/src/test/resources/run:1]: The 140 line of KafkaContainer.java: `URL resource = classLoader.getResource("run");`

File: integration-test/src/test/java/com/datastrato/gravitino/integration/test/web/ui/CatalogsPageKafkaTest.java [hunk 5]
```diff
@@ -0,0 +1,222 @@
...
+            SCHEMA_NAME,
+            TOPIC_NAME);
+    Assertions.assertTrue(catalogsPage.verifyTreeNodes(treeNodes));
+  }
+
+  @Test
+  @Order(3)
+  public void testKafkaTopicDetail() throws InterruptedException {
+    // 1. click toptic tree node
+    String topticNode =
```
Comment (xunliu) [inline integration-test/src/test/java/com/datastrato/gravitino/integration/test/web/ui/CatalogsPageKafkaTest.java:181]: Typo `topticNode` and `// 1. click toptic tree node`

File: integration-test/src/test/java/com/datastrato/gravitino/integration/test/web/ui/CatalogsPageTest.java [hunk 6]
```diff
@@ -377,93 +448,170 @@ public void testRefreshTablePage() {
...
+            TABLE_NAME_2);
     catalogsPage.clickTreeNode(tableNode);
     Assertions.assertTrue(catalogsPage.verifyShowTableTitle(TABLE_TABLE_TITLE));
     Assertions.assertTrue(catalogsPage.verifyShowDataItemInList(COLUMN_NAME_2, true));
     Assertions.assertTrue(catalogsPage.verifyTableColumns());
   }
 
+  @Test
+  @Order(18)
+  public void testOrtherRelationaCatalogTreeNode() throws InterruptedException {
```
Comment (xunliu) [inline integration-test/src/test/java/com/datastrato/gravitino/integration/test/web/ui/CatalogsPageTest.java:523]: Typo `testOrtherRelationaCatalogTreeNode` , `Other`??

**Review by xunliu — APPROVED** [review]
LGTM

**Pull request merged**